packages/esm-patient-forms-app/src/forms/forms-see-all.tsx

```diff
@@ -0,0 +1,128 @@
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { getObjectFHIR } from './get-FHIR-object';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+const Forms = () => {
+  const urlPathArray = window.location.pathname.split('/');
+  const pathUuid = urlPathArray[urlPathArray.length - 2];
+  const [patientUuid, setUUID] = React.useState(null);
+  const [patient, setPatient] = React.useState<fhir.Patient>();
+
+  React.useEffect(() => {
+    getObjectFHIR(patientUuid).then(setPatient);
+    setUUID(pathUuid);
+  }, [pathUuid, patientUuid]);
+
+  const { t } = useTranslation();
+  const displayText = t('forms', 'Forms');
+  const headerTitle = t('forms', 'Forms');
+  const [error, setError] = React.useState(null);
+  const [forms, setForms] = React.useState<Array<Form>>([]);
+  const [encounters, setEncounters] = React.useState<Array<Encounter>>([]);
+  const [completedForms, setCompletedForms] = React.useState<Array<Form>>([]);
+  const [selectedFormView, setSelectedFormView] = React.useState<FormViewState>(FormViewState.all);
+  const [filledForms, setFilledForms] = React.useState<Array<Form>>([]);
+
+  React.useEffect(() => {
+    fetchAllForms().subscribe((forms) => setForms(forms), setError);
+  }, []);
+
+  React.useEffect(() => {
+    const fromDate = dayjs(new Date()).startOf('day').subtract(500, 'day');
+    const toDate = dayjs(new Date()).endOf('day');
+    fetchPatientEncounters(patientUuid, fromDate.toDate(), toDate.toDate()).subscribe(
+      (encounters) => setEncounters(encounters),
+      setError,
+    );
+  }, [patientUuid]);
+
+  React.useEffect(() => {
+    const availableForms = filterAvailableAndCompletedForms(forms, encounters);
+    const completedForms = availableForms.completed.map((encounters) => {
+      encounters.form.complete = true;
+      encounters.form.lastCompleted = encounters.encounterDateTime ? encounters.encounterDateTime : null;
+      return encounters.form;
+    });
+    setCompletedForms(completedForms);
+  }, [forms, encounters]);
+
+  React.useEffect(() => {
+    const filledForms = forms.map((form) => {
+      completedForms.map((completeForm) => {
+        if (completeForm.uuid === form.uuid) {
+          form.complete = true;
+          form.lastCompleted = completeForm.lastCompleted ? completeForm.lastCompleted : null;
+        }
+      });
+      return form;
+    });
+    setFilledForms(filledForms);
```

Comment:
Same here I think this is `useMemo`

Reply:
The explation to this is the same as above and the answer can probably be given by the original committer. Do you think we should go ahead and change all these files too? It would be interesting to find out the logic used though. Should the task of rewrting the whole widget be undertaken now or just try to sort out this minor bug for the time being is a decision to be made. Thanks.

Reply:
We should reuse components - not copy them. This is a no go that results in barely maintainable code that is bloated and contains bugs.

Reply:
As I have explained before the existing component defined in [form-view.component.tsx](https://github.com/openmrs/openmrs-esm-patient-chart/blob/master/packages/esm-patient-forms-app/src/forms/form-view.component.tsx) has hardcoded variables on Ln# 169 and Ln# 170 making reusability beyond scope. Please suggest the right solutions going forward.

Reply:
Even this function gets invoked only once and so probably "useMemo" is not needed.

Reply:
Please read (and understand) my other comment carefully... Thanks!

Reply:
Even here because of the same reason above (since the value of "completedForms" is dependent upon "encounters") the returned value from async "fetch" was not get re-assigned  when useMemo was used. Please let me know if I need to try again and implement as "useMemo" itself.